src/game_event.cpp

```diff
@@ -315,45 +315,41 @@ bool Game_Event::GetActive() const {
 	return data()->active;
 }
 
-void Game_Event::Start(bool by_decision_key) {
+bool Game_Event::SetAsWaitingForegroundExecution(bool face_hero, bool by_decision_key) {
 	// RGSS scripts consider list empty if size <= 1. Why?
-	if (list.empty() || !data()->active)
-		return;
+	if (list.empty() || !data()->active) {
+		return false;
+	}
+
+	if (face_hero && !(IsDirectionFixed() || IsFacingLocked() || IsSpinning())) {
+		SetSpriteDirection(GetDirectionToHero());
+	}
 
-	starting = true;
+	data()->waiting_execution = true;
 	data()->triggered_by_decision_key = by_decision_key;
 	SetPaused(true);
+
+	return true;
 }
 
 const std::vector<RPG::EventCommand>& Game_Event::GetList() const {
 	return list;
 }
 
-void Game_Event::StartTalkToHero() {
-	if (!(IsDirectionFixed() || IsFacingLocked() || IsSpinning())) {
-		SetSpriteDirection(GetDirectionToHero());
-	}
-}
-
 void Game_Event::OnFinishForegroundEvent() {
 	if (!(IsDirectionFixed() || IsFacingLocked() || IsSpinning())) {
 		SetSpriteDirection(GetDirection());
 	}
-
 	SetPaused(false);
-	halting = true;
 }
 
 void Game_Event::CheckEventTriggers() {
 	if (trigger == RPG::EventPage::Trigger_auto_start) {
-		if (Player::GetFrames() == frame_count_at_last_auto_start_check) {
-			// Delay the start to the next frame because the event was enabled
-			// by an event with an higher ID than this event.
-			return;
-		}
-		Start();
+		SetAsWaitingForegroundExecution(false, false);
+		return;
 	} else if (trigger == RPG::EventPage::Trigger_collision) {
 		CheckEventTriggerTouch(GetX(),GetY());
+		return;
 	}
 }
 
```

Comment:
These returns seem unnecessary now.

Reply:
Will fix in #1687 along with the previous comment

Reply:
This call gets blown away in #1697 so I'm leaving it alone for now.